src/data/achievements.js

```diff
@@ -321,6 +326,15 @@ const achievements = [
       return addItemToInventory(state, itemsMap['gold-ingot'], 1, true)
     },
   }))(),
+
+  (() => ({
+    id: 'debt-free',
+    name: 'Debt Free',
+    description: 'Pay off your initial loan from the bank.',
+    rewardDescription: 'Experience bonus',
+    condition: state => state.loanBalance === 0,
+    reward: state => addExperience(state, EXPERIENCE_VALUES.LOAN_PAID_OFF),
```

Comment:
‼️ I found a significant bug with this: Adding experience in this way does not process a level up, so the player does not receive their reward. It seems that `addExperience` should be updated to account for this so that the level-up processing logic is abstracted away. We [discussed this in the previous Experience PR](https://github.com/jeremyckahn/farmhand/pull/448/files/d96a158ee1d599eaeae296be36962eecd1e6e375..fd88b395cd0154059c2dfa2156ea5caa397593ac#r1307529634), but it looks like we'll need to revisit our resolution to that.

I recommend revisiting the implementation of `addExperience` and updating it to process the level-up. We can further discuss how to appropriately test that, but let's settle on a design decision for that first.

Reply:
ahhh goood find. yeah it does feel like `addExperience` should also process level up so that makes sense to me. there are going to be lots of other places where we add experience that are going to need `processLevelUp` called at the same time so it will be easiest if it just does it for us. i'll add that back in

Reply:
added it back in and wrote a test for it based on notifications. open to feedback though of course!